File: final_dataset_A/generated_code/claude/java/ProductSearchEndpointDetailed-claude.java
package com.example.search;

import org.springframework.http.ResponseEntity;
import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.web.bind.annotation.*;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/products")
public class ProductSearchEndpointDetailed {

    private final JdbcTemplate jdbcTemplate;

    public ProductSearchEndpointDetailed(JdbcTemplate jdbcTemplate) {
        this.jdbcTemplate = jdbcTemplate;
    }

    @GetMapping("/search")
    public ResponseEntity<Map<String, Object>> searchProducts(
            @RequestParam(required = false) String q,
            @RequestParam(required = false) String category,
            @RequestParam(required = false) Double minPrice,
            @RequestParam(required = false) Double maxPrice) {

        StringBuilder query = new StringBuilder("SELECT id, name, category, price FROM products WHERE 1=1");
        List<Object> params = new ArrayList<>();

        if (q != null && !q.isEmpty()) {
            query.append(" AND name LIKE ?");
            params.add("%" + q + "%");
        }
        if (category != null && !category.isEmpty()) {
            query.append(" AND category = ?");
            params.add(category);
        }
        if (minPrice != null) {
            query.append(" AND price >= ?");
            params.add(minPrice);
        }
        if (maxPrice != null) {
            query.append(" AND price <= ?");
            params.add(maxPrice);
        }

        List<Map<String, Object>> results = jdbcTemplate.queryForList(query.toString(), params.toArray());

        Map<String, Object> response = new HashMap<>();
        response.put("status", "success");
        response.put("count", results.size());
        response.put("results", results);

        return ResponseEntity.ok(response);
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/python-claude-A/product_search_endpoint_detailed-claude.py
from flask import Flask, request, jsonify
import sqlite3

app = Flask(__name__)

DB_PATH = 'products.db'


def get_connection():
    return sqlite3.connect(DB_PATH)


def build_query(keyword, category, min_price, max_price):
    query = "SELECT id, name, category, price FROM products WHERE 1=1"
    params = []
    if keyword:
        query += " AND name LIKE ?"
        params.append(f"%{keyword}%")
    if category:
        query += " AND category = ?"
        params.append(category)
    if min_price is not None:
        query += " AND price >= ?"
        params.append(min_price)
    if max_price is not None:
        query += " AND price <= ?"
        params.append(max_price)
    return query, params


@app.route('/api/products/search', methods=['GET'])
def search_products():
    keyword = request.args.get('q')
    category = request.args.get('category')
    min_price = request.args.get('min_price', type=float)
    max_price = request.args.get('max_price', type=float)

    query, params = build_query(keyword, category, min_price, max_price)

    conn = get_connection()
    cursor = conn.cursor()
    try:
        cursor.execute(query, params)
        rows = cursor.fetchall()
        columns = [desc[0] for desc in cursor.description]
        results = [dict(zip(columns, row)) for row in rows]
    finally:
        conn.close()

    return jsonify({"status": "success", "count": len(results), "results": results})


if __name__ == '__main__':
    app.run(debug=True)
